From: Patrick Ziegler <zipat@t-online.de>
Subject: [PATCH] Fix makeIntfPair for DockerHost

--- a/mininet/util.py	2019-08-22 08:41:24.821232742 +0200
+++ b/mininet/util.py	2019-08-22 08:43:15.929045810 +0200
@@ -250,18 +250,29 @@
         runCmd( 'ip link del ' + intf1 )
         runCmd2( 'ip link del ' + intf2 )
     # Create new pair
-    netns = 1 if not node2 else node2.pid
+    netns1 = node1.pid
+    netns2 = 1 if not node2 else node2.pid
     if addr1 is None and addr2 is None:
-        cmdOutput = runCmd( 'ip link add name %s '
-                            'type veth peer name %s '
-                            'netns %s' % ( intf1, intf2, netns ) )
+        cmd = 'ip link add name %s netns %s ' \
+              'type veth peer name %s netns %s' \
+              % ( intf1, netns1, intf2, netns2 )
     else:
-        cmdOutput = runCmd( 'ip link add name %s '
-                            'address %s '
-                            'type veth peer name %s '
-                            'address %s '
-                            'netns %s' %
-                            (  intf1, addr1, intf2, addr2, netns ) )
+        cmd = 'ip link add name %s address %s netns %s ' \
+              'type veth peer name %s address %s netns %s' \
+              % ( intf1, addr1, netns1, intf2, addr2, netns2 )
+
+    _, cmdOutput, _ = errRun( cmd )
+
+    # iproute2 changes behaviour in release 5.1
+    # the following workaround should be removed when
+    # issue in iproute2 was fixed
+    # [1] https://github.com/mininet/mininet/issues/884
+    # [2] https://lwn.net/Articles/783494/
+    if "No such device" in cmdOutput:
+        debug( "Ignored error creating interface pair (%s,%s): %s " %
+                         ( intf1, intf2, cmdOutput ) )
+        cmdOutput = ""
+
     if cmdOutput:
         raise Exception( "Error creating interface pair (%s,%s): %s " %
                          ( intf1, intf2, cmdOutput ) )
